rpc/lib/client/ws_client.go

```diff
@@ -175,7 +175,9 @@ func (c *WSClient) OnStart() error {
 }
 
 // OnStop implements cmn.Service.
-func (c *WSClient) OnStop() {}
+func (c *WSClient) OnStop() {
+	_ = c.conn.Close()
```

Comment:
We close connection inside `defer` in readRoutine/writeRoutine. If we don't execute that code, we might have a bigger problem at hand (read/write routines do not exit).

Reply:
@melekes I will take a look

Reply:
@melekes  the problem is likely here https://github.com/tendermint/tendermint/blob/14e7de15dbd01320c7ad146bd02b568e1794fc93/rpc/lib/client/ws_client.go#L477, where it should return, but it does not.

I'm going to close this PR, let me know if that makes sense and I will contribute another one.

Reply:
It looks like it actually should be closed, not sure yet why this is not working.

Reply:
I'm going to test this manually to try and reproduce the issue, as I can see tests there that check this and they seem to pass.

Reply:
Please do. Thank you